assets/js/components/GlobalFilters.vue

```diff
@@ -206,29 +222,47 @@ export default Vue.extend({
       return false;
     },
 
+    removeFilter(e) {
+      const remove_el = e.target.dataset.filter;
+      this.filters[remove_el] = null;
+    },
+    // chronologicalFilters holds the active filters ordered chronologically
     removeLastFilter() {
-      const $this=this;
-      window.addEventListener("keyup", function(e){
-        if (e.keyCode == 27) {
-          if($this.$el.querySelector('.list-filters')){
-          let last_filter = $this.$el.querySelector('.list-filters .filter-item:last-child');
-          if (!last_filter) return;
-          this.filters[last_filter.getAttribute('filter')] = null
-          }
-        }
-      })
+      let removedFilter = this.chronologicalFilters[this.chronologicalFilters.length-1];
```

Comment:
so this isn't a full undo.
why not simply `.pop()`? the last filter still remains in the stack, this doesn't look right.

Reply:
The pop() method removes the last element from an array and returns that element. This method changes the length of the array.
Changes to this.filters call handleFilter, if chronologicalFilters are also handled elsewhere, then in handleFilter they be handled again.
Any changes necessary to chronologicalFilters should be done only in handleFilter.

Reply:
right